### ISSUE TITLE
len-1 scalar is accepted as valid input for len >1 Series

### ISSUE THREAD

**Issue by toobaz** [pandas-dev/pandas#18819]
#### Code Sample, a copy-pastable example if possible

```python
In [2]: pd.Series([1], [0,1])
Out[2]: 
0    1
1    1
dtype: int64
```

#### Problem description

This is undocumented, conceptually wrong, works only because of an implementation glitch, and is used in 3 tests only  - I think - by mistake.

Already discussed (and potentially fixed) inside #18626 (where @jorisvandenbossche suggested a proper deprecation cycle), but @jreback [suggested](https://github.com/pandas-dev/pandas/pull/18769#discussion_r157476271) to open a specific issue.

#### Expected Output

```ValueError: Wrong number of items passed 1, placement implies 2```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: b5f1e716d89487423bb12d6cc4e6da2b39184531
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-4-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.UTF-8
LOCALE: it_IT.UTF-8

pandas: 0.22.0.dev0+371.gb5f1e716d
pytest: 3.2.3
pip: 9.0.1
setuptools: 36.7.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.0dev
tables: 3.3.0
numexpr: 2.6.1
feather: 0.3.1
matplotlib: 2.0.0
openpyxl: 2.3.0
xlrd: 1.0.0
xlwt: 1.3.0
xlsxwriter: 0.9.6
lxml: 4.1.1
bs4: 4.5.3
html5lib: 0.999999999
sqlalchemy: 1.0.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.2.1

</details>

**Comment by jreback** [comment]
yep, this should raise

**Comment by toobaz** [comment]
@jorisvandenbossche true, I was supposed to do so in #18626 , but it was not merged, and a git bisect shows that @cbertinato fixed this instead in #19714, also adding [a test](https://github.com/pandas-dev/pandas/pull/19714/files#diff-3bbe4551f20de6060dce38a95d0adc80R425) (but apparently not a whatsnew)
